### DISCUSSION TITLE
uasyncio create_task methods

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#10415] [Libraries & Drivers] [answered]
Hi,

What is the best way to create a task from a coro and schedule it to run?
Looking at the docs it seems there are 2 ways to do this:
```
# Directly use the create_task from uasyncio
uasyncio.create_task(my_coro())
```
or 
```
# First get the current event_loop instance and use the create_task methof from the returned Loop object.
loop = uasyncio.get_event_loop()
loop.create_task(my_coro())
```

What is the difference between those two methods?
Running `uasyncio.create_task == loop.create_task` returns `False` which means the two functions are not identical apparently...

**Comment by dlech** [comment, marked as answer]
Use `uasyncio.create_task()`. The `Loop.create_task()` method is just a wrapper around it.

https://github.com/micropython/micropython/blob/699477d12d1aa1cb57ff6cfc3f5dc5e97524dbe6/extmod/uasyncio/core.py#L241-L245

**Comment by peterhinch** [comment]
As a general point it is rarely necessary to call `.get_event_loop()`. Those specialised reasons are
 1. If code must be compatible with old versions of CPython.
 2. Creating a global exception handler.

See [uasyncio tutorial](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md).

**Reply by saraverbeecke** [reply]
Great docs again! Thanks for this. I was still using the .get_event_loop, but you are right since there is only one. It also does not make sense anymore to get the loop as the `asyncio.create_task()` handles it already for the (only) active event loop.
